modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/ReplicaTxFinisher.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.partition.replicator;
+
+import static org.apache.ignite.internal.tx.TxState.COMMITTED;
+import static org.apache.ignite.internal.tx.TxState.isFinalState;
+
+import java.util.UUID;
+import org.apache.ignite.internal.hlc.HybridTimestamp;
+import org.apache.ignite.internal.replicator.listener.ReplicaListener;
+import org.apache.ignite.internal.tx.TxManager;
+import org.apache.ignite.internal.tx.TxState;
+import org.apache.ignite.internal.tx.TxStateMeta;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Contains transaction finishing logic for partitions' {@link ReplicaListener} implementations.
+ */
+public class ReplicaTxFinisher {
+    private final TxManager txManager;
+
+    public ReplicaTxFinisher(TxManager txManager) {
+        this.txManager = txManager;
+    }
+
+    /**
+     * Marks the transaction as finished in local tx state map.
+     *
+     * @param txId Transaction id.
+     * @param txState Transaction state, must be either {@link TxState#COMMITTED} or {@link TxState#ABORTED}.
+     * @param commitTimestamp Commit timestamp ({@code null} when aborting).
+     */
+    public void markFinished(UUID txId, TxState txState, @Nullable HybridTimestamp commitTimestamp) {
+        assert isFinalState(txState) : "Unexpected state [txId=" + txId + ", txState=" + txState + ']';
+
+        txManager.updateTxMeta(txId, old -> new TxStateMeta(
+                txState,
+                old == null ? null : old.txCoordinatorId(),
+                old == null ? null : old.commitPartitionId(),
```

Comment:
In `PartitionListener` it was changed from
`old == null ? null : old.commitPartitionId()`
to 
`old == null ? partId : old.commitPartitionId()`
in https://github.com/apache/ignite-3/pull/3591/files#diff-18d5e9727bc86bb1934acb98ca493875900e77fae79fb6c639be87fe9cc3a5d5
I believe this was done on purpose.

Reply:
The code you comment on is from `PartitionReplicaListener`, where the change you mention was NOT done

Reply:
Yes. I missed the fact that for raft you've added RaftTxFinisher.